### PR TITLE
trace:  Span in noop.Start is no longer allocated

### PR DESCRIPTION
**Pull request by tttoad**
trace/noop/noop.Span should be instance. This will reduce memory waste.

Benchmark results

with changes:
new.txt
```
goos: darwin
goarch: arm64
pkg: go.opentelemetry.io/otel/trace/noop
BenchmarkNoopInstance-10        29894822                39.94 ns/op           48 B/op          1 allocs/op
BenchmarkNoopInstance-10        29675424                39.99 ns/op           48 B/op          1 allocs/op
BenchmarkNoopInstance-10        30064700                39.98 ns/op           48 B/op          1 allocs/op
BenchmarkNoopInstance-10        29962016                40.03 ns/op           48 B/op          1 allocs/op
BenchmarkNoopInstance-10        30060465                40.02 ns/op           48 B/op          1 allocs/op
BenchmarkNoopInstance-10        29916855                40.04 ns/op           48 B/op          1 allocs/op
BenchmarkNoopInstance-10        29829998                40.28 ns/op           48 B/op          1 allocs/op
BenchmarkNoopInstance-10        30084706                39.99 ns/op           48 B/op          1 allocs/op
BenchmarkNoopInstance-10        30087441                40.02 ns/op           48 B/op          1 allocs/op
BenchmarkNoopInstance-10        29864365                40.14 ns/op           48 B/op          1 allocs/op
```
without changes on `main`:
old.txt
```
goos: darwin
goarch: arm64
pkg: go.opentelemetry.io/otel/trace/noop
BenchmarkNoopInstance-10        14813442                67.64 ns/op          128 B/op          2 allocs/op
BenchmarkNoopInstance-10        17714486                68.17 ns/op          128 B/op          2 allocs/op
BenchmarkNoopInstance-10        17701257                67.66 ns/op          128 B/op          2 allocs/op
BenchmarkNoopInstance-10        17805859                67.69 ns/op          128 B/op          2 allocs/op
BenchmarkNoopInstance-10        17912841                67.43 ns/op          128 B/op          2 allocs/op
BenchmarkNoopInstance-10        17864120                67.58 ns/op          128 B/op          2 allocs/op
BenchmarkNoopInstance-10        17663130                68.41 ns/op          128 B/op          2 allocs/op
BenchmarkNoopInstance-10        17740423                67.57 ns/op          128 B/op          2 allocs/op
BenchmarkNoopInstance-10        17751040                67.56 ns/op          128 B/op          2 allocs/op
BenchmarkNoopInstance-10        17738064                67.91 ns/op          128 B/op          2 allocs/op

```

benchstat: 
```
goos: darwin
goarch: arm64
pkg: go.opentelemetry.io/otel/trace/noop
                │   old.txt   │               new.txt               │
                │   sec/op    │   sec/op     vs base                │
NoopInstance-10   67.65n ± 1%   40.02n ± 0%  -40.84% (p=0.000 n=10)

                │   old.txt   │              new.txt               │
                │    B/op     │    B/op     vs base                │
NoopInstance-10   128.00 ± 0%   48.00 ± 0%  -62.50% (p=0.000 n=10)

                │  old.txt   │              new.txt               │
                │ allocs/op  │ allocs/op   vs base                │
NoopInstance-10   2.000 ± 0%   1.000 ± 0%  -50.00% (p=0.000 n=10)
```

### CODE REVIEW

**Comment by tttoad** [comment]
```
func BenchmarkNoopInstance(b *testing.B) {
	tracer := NewTracerProvider().Tracer("")
	spanCtx := trace.SpanContext{}

	ctx := trace.ContextWithSpanContext(context.Background(), spanCtx)

	for i := 0; i < b.N; i++ {
		for j := 0; j < 10; j++ {
			newCtx, span := tracer.Start(ctx, "")
			ctx = newCtx
			span.End()
		}
	}
}
// non-single instance
goos: darwin
goarch: arm64
pkg: go.opentelemetry.io/otel/trace/noop
BenchmarkNoopInstance-10         1146366               966.8 ns/op          1280 B/op         20 allocs/op
PASS
ok      go.opentelemetry.io/otel/trace/noop     2.351s

// isingle instance
goos: darwin
goarch: arm64
pkg: go.opentelemetry.io/otel/trace/noop
BenchmarkNoopInstance-10         2038297               555.7 ns/op           480 B/op         10 allocs/op
PASS
ok      go.opentelemetry.io/otel/trace/noop     1.890s
```

**Comment by pellared** [comment]
Looks like https://github.com/open-telemetry/opentelemetry-go/pull/5049

Can you add the benchmark?

**Comment by tttoad** [comment]
@pellared PTAL.

**Review by dmathieu — APPROVED** [review]

File: trace/noop/noop_test.go [hunk 1]
```diff
@@ -101,6 +101,22 @@ func TestTracerStartPropagatesSpanContext(t *testing.T) {
...
 
+func BenchmarkNoopInstance(b *testing.B) {
+	tracer := NewTracerProvider().Tracer("")
+	ctx := trace.ContextWithSpanContext(context.Background(), trace.SpanContext{})
+
+	b.ReportAllocs()
+	b.ResetTimer()
+
+	for i := 0; i < b.N; i++ {
+		for j := 0; j < 10; j++ {
```
Comment (pellared) [inline trace/noop/noop_test.go:112]: Why do we have a second loop?

**Comment by pellared** [comment]
Can you please add a changelog entry similar to https://github.com/open-telemetry/opentelemetry-go/pull/5049 and update the PR description with benchmark results?

File: trace/noop/noop_test.go (hunk 1 above)
Comment (tttoad) [reply trace/noop/noop_test.go]: I was just trying to simulate span being passed in context, but it doesn't need to be. Already removed the.

File: CHANGELOG.md [hunk 2]
```diff
@@ -20,6 +20,7 @@ This project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.htm
 - Identify the `Meter` returned from the global `MeterProvider` in `go.opentelemetry.io/otel/global` with its schema URL. (#5426)
 - Log a warning to the OpenTelemetry internal logger when a `Span` in `go.opentelemetry.io/otel/sdk/trace` drops an attribute, event, or link due to a limit being reached. (#5434)
 - Document instrument name requirements in `go.opentelemetry.io/otel/metric`. (#5435)
+- `Start` in `go.opentelemetry.io/otel/trace/noop` span no longer heap allocated when the parent span is zero SpanContext. (#5457)
```
Comment (MrAlias) [inline CHANGELOG.md:23]:
```suggestion
- `Tracer.Start` in `go.opentelemetry.io/otel/trace/noop` no longer allocates a span for empty span context. (#5457)
```

**Comment by MrAlias** [comment]
Please provide the [benchstat](https://pkg.go.dev/golang.org/x/perf/cmd/benchstat) report in the PR description for the included benchmark.

File: CHANGELOG.md (hunk 2 above)
Comment (pellared) [reply CHANGELOG.md:23]: I suggest also moving it to ` Changed` section (instead of `Fixed`).

**Comment by tttoad** [comment]
@MrAlias @pellared PTAL.

**Comment by MrAlias** [comment]
> @MrAlias @pellared PTAL.

@tttoad where are teh benchstat results?

**Comment by tttoad** [comment]
@MrAlias 
image

I changed the first description of the PR.
Maybe I'm misunderstanding you?

**Comment by tttoad** [comment]
@MrAlias Sorry, first time using `benchstat`. I misunderstood.

**Pull request merged**